solidity/contracts/Router.sol

```diff
@@ -194,31 +202,33 @@ abstract contract Router is HyperlaneConnectionClient, IMessageRecipient {
         uint256 _gasPayment,
         address _gasPaymentRefundAddress
     ) internal returns (bytes32 _messageId) {
-        _messageId = _dispatch(_destinationDomain, _messageBody);
-        // Call the IGP even if the gas payment is zero. This is to support on-chain
-        // fee quoting in IGPs, which should always revert if gas payment is insufficient.
-        interchainGasPaymaster.payForGas{value: _gasPayment}(
-            _messageId,
-            _destinationDomain,
+        // Ensure that destination chain has an enrolled router.
+        bytes32 _router = _mustHaveRemoteRouter(_destinationDomain);
+        bytes memory metadata = GlobalHookMetadata.formatMetadata(
```

Comment:
Tbh I find the name `GlobalHookMetadata` a bit unintuitive. Particularly the "global" bit, which makes sense with the context that before this there used to be an IGP metadata and an OP stack metadata, but without discovering that from https://github.com/hyperlane-xyz/hyperlane-monorepo/pull/2743 I found it confusing

I wonder if something like `HyperlaneHookMetadata` would make sense?

Reply:
sounds good

Reply:
Maybe StandardHookMetadata or something, idk